data/migrations/20190907123916_users.js

```diff
@@ -0,0 +1,42 @@
+exports.up = function(knex) {
+  return knex.schema
+    .createTable("chefs", tbl => {
+      tbl.increments();
+      tbl
+        .text("username", 32)
+        .notNullable()
+        .unique();
+      tbl.text("password").notNullable();
+      tbl
+        .text("name", 32)
+        .notNullable()
+        .unique();
+      tbl.text("city", 64).notNullable();
+      tbl.text("state", 2).notNullable();
```

Comment:
When passing the 2 you will have to make sure the state is always enforced as an abbreviation. That would be handled on the frontend.